Checking for already offered or already chosen driver

Hreherch opened the pull request:
I haven't tested yet, but I think this is a fix. (Squash if you test and agree)

Hreherch committed:
Throw exceptions on important cases for requests

Hreherch committed:
Merge remote-tracking branch 'refs/remotes/origin/master' into offering-requests

Hreherch committed:
Refactored RequestController created AsyncLoad

Hreherch committed:
Fixed listener issue~

Hreherch commented:
Still need to check if the checks are working for offering, but I'm suddenly a lot more relieved.

Hreherch committed:
Moved User Java files to User package

Hreherch committed:
App no longer crashes when making an offer on a searched request

Hreherch committed:
Can no longer make an offer when unneccessary

After hitting the button, or viewing it from MainActivity

Hreherch committed:
Merge remote-tracking branch 'refs/remotes/origin/master' into offering-requests

# Conflicts:

#	app/src/main/java/comcmput301f16t01/github/carrier/Requests/RequestController.java

Hreherch committed:
Logout is now only a menu action

Hreherch commented:
Ready for merge/review
Solves: #148, #134, #131, #151 

@sehoven, @MandyMeindersma, @kieter, @MichaelParadis

Hreherch committed:
Added SwipeRefresh

Also made updating requests more safe (needs to be on UI thread, so use
onPostExecute)

Hreherch committed:
Refresh doesn't block child view now

You can scroll and not be blocked by the refresh now.

Hreherch committed:
Merge pull request #152 from CMPUT301F16T01/swipe-refresh

Swipe Refresh feature

Hreherch commented:
Pull request now also implements code for #127 from merged branch #152.

MandyMeindersma commented on app/src/androidTest/java/comcmput301f16t01/github/carrier/StatusTests.java:
```diff
@@ -5,6 +5,7 @@
 import comcmput301f16t01.github.carrier.Requests.ElasticRequestController;
 import comcmput301f16t01.github.carrier.Requests.Request;
 import comcmput301f16t01.github.carrier.Requests.RequestController;
+import comcmput301f16t01.github.carrier.Users.User;
```
yay beautiful imports

MandyMeindersma commented on app/src/main/java/comcmput301f16t01/github/carrier/MainActivity.java:
```diff
@@ -14,29 +14,35 @@
 import android.support.v7.app.AlertDialog;
 import android.support.v7.app.AppCompatActivity;
 import android.support.v7.widget.Toolbar;
+import android.util.Log;
```
Do we still need logs now? its just debugging?

MandyMeindersma commented on app/src/main/java/comcmput301f16t01/github/carrier/MainActivity.java:
```diff
@@ -284,7 +281,7 @@ public boolean onOptionsItemSelected(MenuItem item) {
      * RiderMainActivity.
      */
-    public void onBackPressed() {
+    public void logout() {
```
update the issue <3

MandyMeindersma commented on app/src/main/java/comcmput301f16t01/github/carrier/MainActivity.java:
```diff
@@ -355,38 +352,88 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container,
+            final SwipeRefreshLayout srl = (SwipeRefreshLayout) rootView.findViewById( R.id.swiperefresh );
+
+            /* src: http://stackoverflow.com/questions/26295481/android-swiperefreshlayout-how-to-implement-canchildscrollup-if-child-is-not-a-l
+             * user: iamlukeyb
```
is this in the wiki?

MandyMeindersma commented on app/src/main/java/comcmput301f16t01/github/carrier/MainActivity.java:
```diff
@@ -355,38 +352,88 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container,
+                public void onRefresh() {
+                    Log.i( "onCreateView", "onRefresh called from SwipeRefreshLayout");
+
+                    if(!ConnectionChecker.isConnected(getContext())) {
```
cool. Good offline stuff

MandyMeindersma commented on app/src/main/java/comcmput301f16t01/github/carrier/MainActivity.java:
```diff
@@ -355,38 +352,88 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container,
+                    ElasticRequestController.setListener(new Listener() {
+                        private int finish = 0;
+                        @Override
+                        public void update() {
```
What is finish? and why is it 2? can we comment here?

MandyMeindersma commented on app/src/main/java/comcmput301f16t01/github/carrier/MainActivity.java:
```diff
@@ -355,38 +352,88 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container,
                 @Override
                 public void update() {
                     requestArrayAdapter.notifyDataSetChanged();
+                    //mSectionsPagerAdapter.notifyDataSetChanged();
```
delete this line now?

MandyMeindersma commented on app/src/main/java/comcmput301f16t01/github/carrier/MainActivity.java:
```diff
@@ -399,7 +446,8 @@ public void update() {
                     Intent intent = new Intent(getActivity(), DriverViewRequestActivity.class);
                     Bundle bundle = new Bundle();
-                    bundle.putString("request", new Gson().toJson(requestList.get(position)));
+                    //bundle.putString("request", new Gson().toJson(requestList.get(position)));
```
delete this now?

MandyMeindersma commented on app/src/main/java/comcmput301f16t01/github/carrier/MainActivity.java:
```diff
@@ -411,27 +459,24 @@ public void onItemClick(AdapterView<?> parent, View view, int position, long id)
                 }
             });
-    
+
```
I AM SARAH DO NOT LEAVE BLANK LINES

MandyMeindersma commented on app/src/main/java/comcmput301f16t01/github/carrier/MainActivity.java:
```diff
@@ -441,8 +486,8 @@ public void update() {
                     Bundle bundle = new Bundle();
-                    bundle.putString("request", new Gson().toJson(requestList.get(position)));
-                    //bundle.putInt( "position", position );
+                    //bundle.putString("request", new Gson().toJson(requestList.get(position)));
```
PLEASE DELETE THIS LINE

MandyMeindersma commented on app/src/main/java/comcmput301f16t01/github/carrier/Requests/DriverViewRequestActivity.java:
```diff
@@ -8,18 +8,18 @@
 import android.widget.TextView;
 import android.widget.Toast;
 
+import com.google.gson.Gson;
```
dem imports ;)

MandyMeindersma commented on app/src/main/java/comcmput301f16t01/github/carrier/Requests/DriverViewRequestActivity.java:
```diff
@@ -62,7 +62,22 @@ protected void onCreate(Bundle savedInstanceState) {
         // unpacking the bundle to get the position of request
         Bundle bundle = getIntent().getExtras();
-        request = new Gson().fromJson(bundle.getString("request"), Request.class);
+        //request = new Gson().fromJson(bundle.getString("request"), Request.class);
```
delete this now

MandyMeindersma commented on app/src/main/java/comcmput301f16t01/github/carrier/Requests/DriverViewRequestActivity.java:
```diff
@@ -319,6 +334,8 @@ public void onClick(DialogInterface dialog, int which) {
         } else {
             rc.addDriver(request, loggedInUser);
             Toast.makeText(this, "Made an offer.", Toast.LENGTH_SHORT).show();
+            view.setEnabled(false); // Make the button un-clickable after offering.
```
great comments

MandyMeindersma commented on app/src/main/java/comcmput301f16t01/github/carrier/Requests/RequestController.java:
```diff
@@ -17,18 +18,45 @@
+    }
+
+    /** Returns an instance of all requests where the user has offered to drive */
+    public RequestList getOffersInstance() {
```
loving the comments

MandyMeindersma commented on app/src/main/java/comcmput301f16t01/github/carrier/Requests/RiderRequestActivity.java:
```diff
@@ -61,10 +55,10 @@ protected void onCreate(Bundle savedInstanceState) {
+        request = rc.getRiderInstance().get(position);
 
-        request = new Gson().fromJson( bundle.getString("request"), Request.class );
+        //request = new Gson().fromJson( bundle.getString("request"), Request.class );
```
delete this line?

MandyMeindersma commented on app/src/main/java/comcmput301f16t01/github/carrier/Searching/SearchResultsActivity.java:
```diff
@@ -36,6 +36,7 @@ protected void onCreate(Bundle savedInstanceState) {
             public void onItemClick(AdapterView<?> parent, View view, int position, long id) {
                 Intent intent = new Intent(SearchResultsActivity.this, DriverViewRequestActivity.class);
                 Bundle bundle = new Bundle();
+                bundle.putInt( "position", -1 );
```
? what does this do? add a comment?

MandyMeindersma reviewed (changes requested):
Once you look at all these comments could you see what was up with that weird crashing error. 

Other than that you did a wonerful amazing job and I am so proud of you

Hreherch committed:
Fixed missing offeredRequests issue

Hreherch committed:
Better documentation for MainActivity

Hreherch committed:
Added request filtering when using search by keyword

Extracted subtasks in ElasticRequestController and made them avaliable
to any async task that uses them.

Hreherch committed:
Commenting

Hreherch committed:
Merge remote-tracking branch 'refs/remotes/origin/master' into offering-requests

# Conflicts:

#	app/src/main/java/comcmput301f16t01/github/carrier/Requests/DriverRequestAdapter.java

Also removed a commented line

Hreherch commented:
@MandyMeindersma I'm not going to fix the space because that's a simple refactor right click action and we can do that at the end and I don't want merge conflicts. But I think I'm done, test it out!

Hreherch committed:
Should have been deleted in the conflict
